test/tests/functional/pbs_calendaring.py

```diff
@@ -191,3 +191,45 @@ def test_topjob_fail(self):
 
         msg = jid3 + ';Job is a top job and will run at'
         self.scheduler.log_match(msg)
+
+    @skipOnCpuSet
+    def test_topjob_bucket(self):
+        """
+        In this test we test that a bucket job will be calendared to start
+        at the end of the last job on a node
+        """
+
+        self.scheduler.set_sched_config({'strict_ordering': 'true all'})
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        res_req = {'Resource_List.select': '1:ncpus=1',
+                   'Resource_List.walltime': 30}
+        j1 = Job(TEST_USER, attrs=res_req)
+        j1.set_sleep_time(30)
+        jid1 = self.server.submit(j1)
+
+        res_req = {'Resource_List.select': '1:ncpus=1',
+                   'Resource_List.walltime': 45}
+        j2 = Job(TEST_USER, attrs=res_req)
+        j2.set_sleep_time(45)
+        jid2 = self.server.submit(j2)
+
+        res_req = {'Resource_List.select': '1:ncpus=1',
+                   'Resource_List.place': 'excl'}
+        j3 = Job(TEST_USER, attrs=res_req)
+        jid3 = self.server.submit(j3)
+
+        self.server.expect(JOB, {'job_state': 'R'}, jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, jid3)
+        job1 = self.server.status(JOB, id=jid1)
+        job2 = self.server.status(JOB, id=jid2)
+        job3 = self.server.status(JOB, id=jid3)
+
+        stime = time.strptime(job2[0]['stime'], "%a %b %d %H:%M:%S %Y")
```

Comment:
If you don't want to jump through all the hoops of using strptime() and mktime(), you can use the log utils convert_datetime() function.  It takes an optional format.  Also, consider using %c instead of the longer format string.

Reply:
I'd rather use strptime and mktime, in case convert_date_time changes I really don't want to have to deal with changing this later.
I will switch to using %c though